pkg/cloud/services/networking/service.go

```diff
@@ -81,6 +81,11 @@ func NewTestService(projectID string, client NetworkClient, logger logr.Logger)
 // replaceAllAttributesTags replaces all tags on a neworking resource.
 // the value of resourceType must match one of the allowed constants: trunkResource or portResource.
 func (s *Service) replaceAllAttributesTags(eventObject runtime.Object, resourceType string, resourceID string, tags []string) error {
+	if len(tags) == 0 {
+		s.logger.Info("IgnoreReplaceAllAttributesTags", "no tags provided: , %s", resourceID)
```

Comment:
nit: I suspect this log isn't quite right, but I don't think it's important enough to hold up the fix.

/lgtm

Reply:
What should change it to ?
I can make the changes on another (ongoing PR) which also related to trunk.

Reply:
I don't think we're using this consistently at the moment which is why I only called this a nit, but I don't think the `%s` will be expanded in the second argument as you intend because this is a key/value pair. IIUC (and I'm not 100% on this without testing), it should be something like:

```golang
s.logger.Info("no tags provided to ReplaceAllAttributesTags")
```

Looking at how we're initialising the logger, I don't think we even need to log the resource ID because the machine name is already included:

https://github.com/kubernetes-sigs/cluster-api-provider-openstack/blob/2a51dde1157f259efce1569c7901ceef27954c7b/controllers/openstackmachine_controller.go#L89

and

https://github.com/kubernetes-sigs/cluster-api-provider-openstack/blob/2a51dde1157f259efce1569c7901ceef27954c7b/controllers/openstackcluster_controller.go#L91

IIUC this log message is just going to look ugly with an un-expanded string in it. I'm still happy to accept a follow-up PR as I doubt it would confuse anybody in practise and it fixes a real bug.

/lgtm